src/arviz_plots/plots/evolutionplot.py

```diff
@@ -0,0 +1,449 @@
+        if "model" in distribution:
+            pc_kwargs["aes"].setdefault("color", ["model"])
+            # setting x aesthetic to np.linspace(-x_diff/3, x_diff/3, length of 'model' dim)
+            # x_diff = span of x axis (1) divided by number of points to be plotted (n_points)
+            x_diff = 1 / n_points
+            if "x" not in pc_kwargs:
+                pc_kwargs["x"] = np.linspace(-x_diff / 3, x_diff / 3, distribution.sizes["model"])
+            pc_kwargs["aes"].setdefault("x", ["model"])
+        aux_dim_list = [dim for dim in pc_kwargs["cols"] if dim != "__variable__"]  # for divergence
+        plot_collection = PlotCollection.wrap(
+            distribution,
+            backend=backend,
+            **pc_kwargs,
+        )
+    else:
+        aux_dim_list = list(
+            set(
+                dim for child in plot_collection.viz.children.values() for dim in child["plot"].dims
+            )
+        )
+
+    # set plot collection dependent defaults (like aesthetics mappings for each artist)
+    if aes_map is None:
+        aes_map = {}
+    else:
+        aes_map = aes_map.copy()
+    aes_map.setdefault("bulk", plot_collection.aes_set.difference({"overlay"}))
+    aes_map.setdefault("tail", plot_collection.aes_set.difference({"overlay"}))
+    aes_map.setdefault("mean", plot_collection.aes_set)
+    aes_map.setdefault("mean", ["linestyle", "-"])
+    aes_map.setdefault("sd", plot_collection.aes_set)
+    aes_map.setdefault("sd", ["linestyle", "-"])
+    aes_map.setdefault("divergence", {"overlay"})
+    if labeller is None:
+        labeller = BaseLabeller()
+
+    # compute and add ess evolution subplots for 'bulk' and 'tail'
+
+    n_draws = distribution.sizes["draw"]
+    n_samples = n_draws * distribution.sizes["chain"]
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+    default_bulk_color = plot_bknd.get_default_aes("color", 2, {})[0]
+    default_tail_color = plot_bknd.get_default_aes("color", 2, {})[1]
+
+    bulk_kwargs = copy(plot_kwargs.get("bulk", {}))
+
+    if bulk_kwargs is not False:
+        # step 2
+        bulk_dims, bulk_aes, bulk_ignore = filter_aes(plot_collection, aes_map, "bulk", sample_dims)
+        first_draw = distribution.draw.values[0]
+        ylabel = "{}"
+        xdata = np.linspace(n_samples / n_points, n_samples, n_points)
+        draw_divisions = np.linspace(n_draws // n_points, n_draws, n_points, dtype=int)
+        ess_y_dataset = xr.concat(
+            [
+                distribution.sel(draw=slice(first_draw + draw_div)).azstats.ess(
+                    dims=bulk_dims,
+                    method="bulk",
+                    relative=relative,
+                    **stats_kwargs.get("ess_bulk", {}),
+                )
+                for draw_div in draw_divisions
+            ],
+            dim="ess_dim",
+        )
+
+        # ----------------------------
+
+        xdata = np.linspace(0, n_samples, n_points)
```

Comment:
This is overwriting the xdata above but I think we want the one above, so I'd remove this line. Otherwise we get ess not necessarly 0 at 0 samples which makes no sense